File: pyg_1_3/pyg_parent/pyg_cart_web/src/main/webapp/js/service/payService.js
//支付服务层
app.service('payService', function ($http) {
    // 生成微信支付url
    this.createNative = function () {
        return $http.get('../pay/createNative.do');
    };
    // 检查支付状态
    this.queryPayStatus = function (out_trade_no) {
        return $http.get('../pay/queryPayStatus.do?out_trade_no='+out_trade_no);
    };
});

File: pyg_1_3/pyg_parent/pyg_pay_interface/src/main/java/com/pyg/pay/service/PayService.java
package com.pyg.pay.service;

import java.util.Map;

/**
 * 支付服务接口
 *
 * @author 杨立波  2018-09-22 16:56
 */

public interface PayService {

    /**
     * 调用微信支付api，生成微信扫码支付url
     *
     * @param out_trade_no 商户订单号
     * @param total_fee 标价金额
     * @return Map
     */
    public Map createNative(String out_trade_no, String total_fee);

    /**
     * 查询订单支付状态
     *
     * @param out_trade_no 商户订单号
     * @return Map
     */
    public Map queryPayStatus(String out_trade_no);

    /**
     * 关闭支付
     *
     * @param out_trade_no 商户订单号
     * @return Map
     */
    public Map closePay(String out_trade_no);
}
